src/Camera/Camera.h

```diff
@@ -0,0 +1,49 @@
+/* Copyright (c) 2018 mickael.leclerc@epitech.eu charles.fraisse@epitech.eu
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+ The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
+OR OTHER DEALINGS IN THE SOFTWARE. */
+
+#pragma once
+
+#include <array>
+#include "../Vector/Vector3.h"
+#include "../Vector/Vector2.h"
+
+namespace rt {
+class Camera {
+ public:
+    Camera(Vector3 const& pos, Vector3 const& target);
+    explicit Camera(Vector3 const& target);
+    virtual ~Camera(void);
+
+    Vector3  generateRay(Vector2 const &pos) const;
+    void     generateAxis(Vector3 const& target);
+    void     generateScreen();
+
+    std::array<Vector3, 3> const& getAxis(void) const;
+    Vector2 const&                getRes(void) const;
+    void                          setRes(const Vector2& res);
+
+ private:
+    Vector3                 _pos = Vector3(0, 0, 0);
+    std::array<Vector3, 3>  _axis;
+    float                   _fov = 90.0;
+    Vector2                 _screenRes = Vector2(1920, 1080);

```

Comment:
Screen res will be defined before generating the camera, meaning that the camera need to be created with a res parameter